devnotes.md

```diff
@@ -1,44 +1,89 @@
-### Build
-`npx hardhat compile`
+## Build
 
+To compile the project, run the following command:
 
-### Deploying
-
-Set your private key in `.env`:
+```bash
+npx hardhat compile
 ```
+
+
+## Deploying
+
+### Set Up Environment
+
+Ensure your private key is set in the `.env` file:
+
+```bash
 PRIVATE_KEY="..."
 ```
 
-Deploy the contract
-```
+### Deploy the Contract
+
+To deploy the contract, execute:
+
+```bash
 npx hardhat ignition deploy ignition/modules/F3Parameters.ts --network filecoinmainnet \
-   --parameters ignition/parameters_mainnet.json --deployment-id mainnet
+  --parameters ignition/parameters_mainnet.json --deployment-id mainnet
 ```
 
-### Validating Contract
+## Validating Contract
 
-Go to BlockScout contract page, verify, then get select `Solidity (Standard JSON Input)`
-run:
-`jq .input ignition/deployments/chain-314/build-info/[there is single json file here] > validation-input.json`
+1. Navigate to the BlockScout contract page and verify the contract.
+2. Select `Solidity (Standard JSON Input)`.
+3. Run the following command to prepare the validation input:
 
-Upload this file to BlockScout
+   ```bash
+   jq .input ignition/deployments/chain-314/build-info/[there is single json file here] > validation-input.json
+   ```
 
-### Validating parameters proposed in Safe Wallet
+4. Upload `validation-input.json` to BlockScout.
 
-In Safe UI you will see a proposed message.
-You can use `npx hardhat validateActivationMessage` to cross-validate it with manifest
-specified in this repository.
+### End-to-End Testing with Node Implementation
 
+This section describes how to perform end-to-end testing of the contract using a node implementation. The node implementation includes similar functionality to `queryActivationInformation` but operates on a polling basis.
 
-Example:
-If the Safe call is to contract `0xD68211Bf70c9791AE1c5307970D3ca9a43834B76` with hex encodede data of:
-```
+#### Deployment
+
+1. **Set Up Environment**: Ensure your private key is set in the `.env` file:
+   ```
+   PRIVATE_KEY="..."
+   ```
+
+2. **Deploy the Contract**: Deploy the contract using your private key. It is recommended to deploy on the mainnet for best results with BlockScout. Avoid specifying the parameters file to use the deploying key as the owner:
+   ```bash
+   npx hardhat ignition deploy ignition/modules/F3Parameters.ts --network filecoinmainnet --deployment-id mainnet-scratchpad1
```

Comment:
```suggestion
   npx hardhat ignition deploy ignition/modules/F3Parameters.ts --network filecoinmainnet --deployment-id mainnet-scratchpad-$USER
```

Just to make it clear that this is a user-focused scratchpad?  (I initially missed the 1 suffix.)

Reply:
personally I don't love it, but I guess it can work